app/lib/actions/candlepin/reindex_pool_subscription_handler.rb

```diff
@@ -44,24 +44,25 @@ def handle(message)
 
       private
 
-      def reindex_or_unindex_pool(pool_id)
-        ::Katello::Pool.find_pool(pool_id)
-        reindex_pool(pool_id)
+      def import_or_remove_pool(pool_id)
+        ::Katello::Pool.find_by_cp_id(pool_id).import_data
       rescue RestClient::ResourceNotFound
-        unindex_pool(pool_id)
+        remove_pool(pool_id)
       end
 
-      def reindex_pool(pool_id)
-        @logger.info "re-indexing pool #{pool_id}."
-        pool = ::Katello::Pool.find_pool(pool_id)
-        ::Katello::Pool.index_pools([pool]) unless pool.nil? || pool.unmapped_guest
+      def import_pool(pool_id)
+        ::Katello::Pool.find_or_create_then_index(pool_id)
       rescue RestClient::ResourceNotFound
         @logger.debug "skipped re-index of non-existent pool #{pool_id}"
       end
 
-      def unindex_pool(pool_id)
-        @logger.info "removing pool from index #{pool_id}."
-        ::Katello::Pool.remove_from_index(pool_id)
+      def remove_pool(pool_id)
+        pool = ::Katello::Pool.find_by_cp_id(pool_id)
+        if pool
+          pool.destroy!
+        else
+          @logger.debug "Couldn't find pool with candlepin id #{pool_id} in the database"
```

Comment:
Is that handled here? https://github.com/Katello/katello/blob/master/app/lib/actions/candlepin/reindex_pool_subscription_handler.rb#L25